### ISSUE TITLE
[Question]: sphinxcontrib-applehelp require (python>=3.9, sphinx >=5.0) leads to fail of "Test code" workflow with sphinx-4

### ISSUE THREAD

**Issue by minhdoan0730** [rstcheck/rstcheck-core#89]
### To Dos

- [X] I already searched with my favorite search engine "How to do X with rstcheck".
- [X] I already read and followed the README and didn't find an answer.
- [X] I have checked the [issues](https://github.com/rstcheck/rstcheck/issues) and think that this is not a duplicate.

- [X] I added a very descriptive title to this issue.

### Description

Hello, I'm a developer who work with [Roberto](https://github.com/r-o-b-e-r-t-o) to implement feature https://github.com/rstcheck/rstcheck-core/issues/68.
Our pull request here https://github.com/rstcheck/rstcheck-core/pull/85.

Currently the Test code CI workflow of our PR is failed with the python version >=3.9 in matrix. As the logs show while running **tox**, it's related to sphinxcontrib.applehelp that require python>=3.9 and sphinx>=5.0 https://github.com/sphinx-doc/sphinxcontrib-applehelp/blob/master/pyproject.toml#L54

```
.tox/py310-with-sphinx4/lib/python3.10/site-packages/sphinx/application.py:380: in setup_extension
    self.registry.load_extension(self, extname)
        extname    = 'sphinxcontrib.applehelp'
        self       = <sphinx.application.Sphinx object at 0x103eaa110>
.tox/py310-with-sphinx4/lib/python3.10/site-packages/sphinx/registry.py:441: in load_extension
    raise VersionRequirementError(
E   sphinx.errors.VersionRequirementError: The sphinxcontrib.applehelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.
```

I also take the test of removing sphinx-4 from tox-env on my personal fork, it all passed, https://github.com/namminhlp/rstcheck-core/actions/runs/7608363389.

So, is the issue come from your setup ? Or what else we need to do to pass your Test code workflow ?

Thank you in advance !

### Operating System

Linux, macOS

### Python Version

3.11

### rstcheck Version

1.2.2

**Comment by Cielquan** [comment]
Hi, I'll have to take a closer look at this. Right off the bat I can't tell much and nothing rings a bell at sphinxcontrib.applehelp. I hope I find the time in the next few days.

**Comment by Cielquan** [comment]
Just saw other PRs failing with the same error, also over at `rstcheck`. Seems to be a bigger issue not only for your PR.

**Comment by Cielquan** [comment]
Fixed in #90

**Comment by minhdoan0730** [comment]
Thank you @Cielquan
